torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -970,9 +970,16 @@ std::exception_ptr ProcessGroupNCCL::checkForNCCLErrorsInternal(
     }
     ncclResult_t ncclAsyncErr = ncclComm->checkForNcclError();
     if (ncclAsyncErr != ncclSuccess) {
+      std::string additionalMsg;
+      if (ncclAsyncErr == ncclSystemError) {
+        additionalMsg = c10::str("\n",
```

Comment:
Probably better to extend the error message directly here: https://github.com/pytorch/pytorch/blob/master/torch%2Fcsrc%2Fdistributed%2Fc10d%2FNCCLUtils.hpp#L28

Reply:
Appreciate your comment @pritamdamania87 ! 
I put the additional message here because I need to rely on the fact that this is an **asynchronous** NCCL error (i.e. dynamic during a NCCL operation), rather than applying the hint to all ncclSystemError's which may also include those immediately returned during NCCL initialization. Errors like "connection closed by remote peer" most happen asynchronously during NCCL execution.

Reply:
Updated code change to reflect @pritamdamania87 's comment.